tools/clang/lib/Sema/SemaDecl.cpp

```diff
@@ -5820,7 +5820,7 @@ Sema::ActOnVariableDeclarator(Scope *S, Declarator &D, DeclContext *DC,
 
         // C++98 [class.union]p1: If a union contains a static data member,
         // the program is ill-formed. C++11 drops this restriction.
-        if (RD->isUnion())
+        if (RD->isUnion() && !getLangOpts().HLSL)
```

Comment:
Something else that occurred to me. Is this change right? Do we want to allow static data members inside a union? Does that even make sense?

Reply:
Yes, Standard says that static data members are allowed inside a union.